black.py

```diff
@@ -618,7 +636,8 @@ def format_str(
         remove_u_prefix=py36 or "unicode_literals" in future_imports,
         is_pyi=is_pyi,
         normalize_strings=normalize_strings,
-        allow_underscores=py36,
+        allow_underscores=py36
+        and not bool(mode & FileMode.NO_NUMERIC_UNDERSCORE_NORMALIZATION),
```

Comment:
Wow this looks ugly

Reply:
I personally prefer the optional parentheses even if it makes the expression break over 4 lines. Not sure if that is the best way to go in the general case, though.